### ISSUE TITLE
[FEATURE] document response in depends

### ISSUE THREAD

**Issue by Trim21** [tiangolo/fastapi#501]
**Is your feature request related to a problem? Please describe.**
No not related to a problem.

I'm using ` fastapi.security.api_key.APIKeyCookie` as a depends and it may raise `HTTPException(403, detail="Not authenticated")` for any router it is used. So i have to document a 403 responses in all related router.

**Describe the solution you'd like**

Could there by a way when I'm using a class as Depends, document responses once as a class member or some thing else instead of document it many times in routers.

**Comment by prostomarkeloff** [comment]
But... why don't you just add this response in router registration?

It should look like that:
```py
# ...
app.include_router(my_router, responses={403: ...})
```

**Comment by Trim21** [comment]
@prostomarkeloff Like why using depends instead of calling it in handlers...

**Comment by prostomarkeloff** [comment]
> @prostomarkeloff Like why using depends instead of calling it in handlers...

It doesn't look equal 😂

**Comment by Trim21** [comment]
@prostomarkeloff Sorry, my apology.

Some of my routers from one APIRouter are not using same depdencies.

**Comment by tiangolo** [comment]
Yep, I want to have this in some way. But I have to figure out some other features and changes first. But I want to solve this in some way. 🤓 

> Sorry for the long delay! 🙈  I wanted to personally address each issue and they piled up through time, but now I'm checking each one in order.

**Comment by iudeen** [comment]
@tiangolo , if you or anyone hasn't got a chance to look into this, shall I take a stab at this?

**Comment by trim21** [comment]
have been waiting for progress for 3 years…

**Comment by codespearhead** [comment]
I'm not sure I understand the issue, but won't a Custom Exception Handler do the trick?

```py
from fastapi import Depends, HTTPException, APIRouter, status

class NotAuthenticatedException(HTTPException):
    def __init__(self, detail="Not authenticated"):
        super().__init__(status_code=status.HTTP_403_FORBIDDEN, detail=detail)

router = APIRouter()

@router.get("/protected-endpoint")
async def protected_endpoint(api_key: str = Depends(APIKeyCookie(secret="your_secret_key"))):
    print("Hello world")
```

**Comment by trim21** [comment]
> I'm not sure I understand the issue, but won't a Custom Exception Handler do the trick?
> 
> ```python
> from fastapi import Depends, HTTPException, APIRouter, status
> 
> class NotAuthenticatedException(HTTPException):
>     def __init__(self, detail="Not authenticated"):
>         super().__init__(status_code=status.HTTP_403_FORBIDDEN, detail=detail)
> 
> router = APIRouter()
> 
> @router.get("/protected-endpoint")
> async def protected_endpoint(api_key: str = Depends(APIKeyCookie(secret="your_secret_key"))):
>     print("Hello world")
> ```

No, you didn't understand this issue. It's not about how to raise a exception or which exception to raise, but how to add a 403 response in openapi for all endpoint using a `Depends`, for example,  `APIKeyCookie`

**Comment by code-yeongyu** [comment]
.

**Comment by code-yeongyu** [comment]
.

**Comment by AlexeyReket** [comment]
Any progress in this issue?
I have the same need to modify openapi from depends😅
Can i or someone else help you to make this possible?🙌

**Comment by tiangolo** [comment]
This is something I've considered for a while, but there are several other priorities first.

I'm still tracking that internally, for now I'll close this specific issue.

---

Having issues open for specific things is becoming problematic as they attract AI slop PRs, so I'm closing this for cleanup. ☕
